### ISSUE TITLE
[Android] Master-Detail-Page

### ISSUE THREAD

**Issue by kulbe-redlogix** [alexrainman/CarouselView#350]
We are using the CarouselView with a Master-Detail-Page Navigation. The problem is if you change the view from the NavigationPage with the CarouselView to another Page and then back to the Carousel Page again, the CarouselView will not be visible.

We found the problem in Element_SizeChanged, the variable canSetLayout is false and therefore the view will not be rendered again. The check for the layout was made because of Issue #136 that the page is recreated multiple times. We deleted the check and the Renderer is working as intended.

I debugged the code with a Master-Detail-Page Navigation.
At the first time the method Element_SizeChanged is called three times:
1. OnAttachedWindow (canSetLayout: false)
- MainPageSizeChanged will be set canSetLayout back to false
2. ElementSizeChanged (canSetLayout: false)
3. ElementSizeChanged (canSetLayout: true) --> here the View is rendered initially
If the Master-Detail-Page is changing to another View the CarouselViewRenderer will be disposed. After switching back to the CarouselView again, a new CarouselViewRenderer will be created and the ElementSizeChanged is called only once (canSetLayout false).

**Comment by tmt242001** [comment]
I think this is an important fix.

**Comment by alexrainman** [comment]
This control has been merged into Xamarin.Forms master branch. Please, wait for it to be released as part of Xamarin.Forms and submit any issues there https://github.com/xamarin/xamarin.forms/tree/feature-carouselview. Thanks
